src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -93,9 +95,106 @@ internal abstract class AbstractRecommendationService : IRecommendationService
                 Concat(symbols.WhereAsArray(s => s.Kind != SymbolKind.Field));
         }
 
-        private static bool IsFriendlyName(int i, string elementName)
+        protected static ImmutableArray<ISymbol> LookupSymbolsInContainer(
+            INamespaceOrTypeSymbol container,
+            SemanticModel semanticModel,
+            int position,
+            bool excludeInstance)
         {
-            return elementName != null && string.Compare(elementName, "Item" + (i + 1), StringComparison.OrdinalIgnoreCase) != 0;
+            return excludeInstance
+                    ? semanticModel.LookupStaticMembers(position, container)
+                    : SuppressDefaultTupleElements(container,
+                        semanticModel.LookupSymbols(position, container, includeReducedExtensionMethods: true));
+        }
+
+        protected static ImmutableArray<ISymbol> GetSymbols(
+            SemanticModel semanticModel,
+            INamespaceOrTypeSymbol container,
+            int position,
+            bool useBaseReferenceAccessibility,
+            bool excludeInstance)
+        {
+            return useBaseReferenceAccessibility
+                ? semanticModel.LookupBaseMembers(position)
+                : LookupSymbolsInContainer(container, semanticModel, position, excludeInstance);
+        }
+
+        protected static ImmutableArray<ISymbol> GetSymbols<TLambdaExpressionSyntax, TArgumentSyntax, TArgumentListSyntax, TInvocationExpressionSyntax>(
+            SemanticModel semanticModel,
+            IParameterSymbol parameter,
+            int position,
+            Func<TArgumentListSyntax, SeparatedSyntaxList<TArgumentSyntax>> getArguments,
+            CancellationToken cancellationToken)
+            where TArgumentListSyntax : SyntaxNode
+            where TArgumentSyntax : SyntaxNode
+            where TInvocationExpressionSyntax : SyntaxNode
+            where TLambdaExpressionSyntax : SyntaxNode
+
+        {
+            if (!(parameter.ContainingSymbol is IMethodSymbol containingMethod &&
```

Comment:
this whole method needs comments helping explain what is going on.  i.e.: `// this code is to help give intellisense in the following case: ...`.  And then explain what we're looking for and what this is working around.

Reply:
i'm going to hold off until then, aas that will really help me understand what's going on here and why it's appropriate.  Ping me whne you think it's good to look again!